### ISSUE TITLE
Navigation block: aria-label disappears after any change to the pattern the block belongs to

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#68719]
### Description

The navigation block doesn't preserve the default aria-label attributes passed from a thene pattern when making any change to the template part the pattern is used into.

As such, the navigation `<nav>` elements become unexpectedly unlabeled.
Multiple `<nav>` elements must be labeled to allow users to distinguish what each navigation is about.
Loss of content should be prevented.

### Step-by-step reproduction instructions

- Activate Twenty Twenty-Four.
- Make sure the Footer hasn't been customized. Reset any change if necessary.
- Go to any page on the front end.
- Inspect the 3 `<nav>` elements in the footer.
- Observe they all have an appropriate aria-label: About, Privacy, Social Media.
- Note: these are the values that come from [the theme pattern footer.php](https://github.com/WordPress/wordpress-develop/blob/5b01d24d8c5f2cfa4b96349967a9759e52888d03/src/wp-content/themes/twentytwentyfour/patterns/footer.php).
- Edit the footer and make any unrelated change, e.g.: change the Site title to open in a new tab and save.

![Image](https://github.com/user-attachments/assets/f1237912-bc6b-43fb-ac21-0127db4d2a4a)

- Go to any page on the front end.
- Inspect the 3 `<nav>` elements in the footer.
- Observe they don't have an aria-label any longer.
- Expected: to preserve the original labels: About, Privacy, Social Media.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by im3dabasia** [comment]
Hi @afercia ,

Thank you for raising the issue. I was able to reproduce it . The aria-labels do not persist after making changes to the footer. 

<img width="1468" alt="Image" src="https://github.com/user-attachments/assets/571765f5-0970-4dfe-a8f4-eac1423b0481" />

**Comment by t-hamano** [comment]
I think the underlying problem is that the `ariaLabel` block support doesn't work with dynamic blocks. See https://github.com/WordPress/gutenberg/issues/68764.

**Comment by Mamaduka** [comment]
Should we close specific issues with `ariaLabel` block support in favor of a new general issue?
